electrum/tests/test_transaction.py

```diff
@@ -917,3 +917,145 @@ def test_spending_op_cltv_p2wsh(self):
                          tx.txid())
         self.assertEqual('020000000001019ad573c69e60c209e0ff36f281ae4f700a8d59f846e7ff5c020352fd1e97808600000000000000000001fa840100000000001600145a209b202bc19b3d345a75cf8ab51cb471913a790247304402207b191c1e3ff1a2d3541770b496c9f871406114746b3aa7347ec4ef0423d3a975022043d3a746fa7a794d97e95d74b6d17d618dfc4cd7644476813e08006f271e51bd012a046c4f855fb1752102aec53aa5f347219a7378b13006eb16ce48125f9cf14f04a5509a565ad5e51507ac6c4f855f',
                          tx.serialize())
+
+class Test_Sighash_Types(ElectrumTestCase):
+    #These tests are taken from bip143, https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki
+    def check_sighash_types_sighash_all(self):
+        #Input of transaction
+        locktime=0
```

Comment:
There's lots of code repetition, it would be cool if you could define the transaction in one place and then only change the sighash types.